performance-tests/src/main/java/io/kroxylicious/benchmarks/InvokerDispatchBenchmark.java

```diff
@@ -0,0 +1,188 @@
+                    invokerType.invokerWith(new FourInterfaceFilter0()),
+                    invokerType.invokerWith(new FourInterfaceFilter1()),
+                    invokerType.invokerWith(new FourInterfaceFilter2()),
+                    invokerType.invokerWith(new FourInterfaceFilter3()),
+                    invokerType.invokerWith(new FourInterfaceFilter0()),
+                    invokerType.invokerWith(new FourInterfaceFilter1()),
+                    invokerType.invokerWith(new FourInterfaceFilter2()),
+                    invokerType.invokerWith(new FourInterfaceFilter3())
+            };
+            final Map<ApiKeys, ApiMessage> messages = Map.of(ApiKeys.PRODUCE, new ProduceRequestData(), ApiKeys.API_VERSIONS, new ApiVersionsRequestData(), ApiKeys.FETCH,
+                    new FetchRequestData());
+            apiMessages = messages.entrySet().toArray(new Map.Entry[0]); // Avoids iterator.next showing up in the benchmarks
+            requestHeaders = new RequestHeaderData();
+            filterContext = new StubFilterContext();
+            keys = messages.keySet().toArray(new ApiKeys[0]);
+        }
+    }
+
+    @Benchmark
+    public void testDispatchToShouldHandle(BenchState state, Blackhole blackhole) {
+        invokeShouldHandle(blackhole, state.invokers, state.keys);
+    }
+
+    @Benchmark
+    public void testDispatchToHandleRequest(BenchState state) {
+        invokeHandleRequest(state.invokers, state.apiMessages, state.requestHeaders, state.filterContext);
+    }
+
+    @Benchmark
+    @Threads(4)
+    public void test4ThreadsDispatchToShouldHandle(BenchState state, Blackhole blackhole) {
+        invokeShouldHandle(blackhole, state.invokers, state.keys);
+    }
+
+    @Benchmark
+    @Threads(4)
+    public void test4ThreadsDispatchToHandleRequest(BenchState state) {
+        invokeHandleRequest(state.invokers, state.apiMessages, state.requestHeaders, state.filterContext);
+    }
+
+    private static void invokeShouldHandle(Blackhole blackhole, FilterInvoker[] filters, ApiKeys[] apiKeys) {
+        for (ApiKeys apiKey : apiKeys) {
+            final short apiVersion = apiKey.latestVersion();
+            for (FilterInvoker invoker : filters) {
+                blackhole.consume(invoker.shouldHandleRequest(apiKey, apiVersion));
+                blackhole.consume(invoker.shouldHandleResponse(apiKey, apiVersion));
+            }
+        }
+    }
+
+    private static void invokeHandleRequest(FilterInvoker[] filters, Map.Entry<ApiKeys, ApiMessage>[] apiMessages, RequestHeaderData requestHeaders,
+                                            KrpcFilterContext filterContext) {
+        for (Map.Entry<ApiKeys, ApiMessage> entry : apiMessages) {
+            final ApiKeys apiKey = entry.getKey();
+            final short apiVersion = apiKey.latestVersion();
+            for (FilterInvoker invoker : filters) {
+                if (invoker.shouldHandleRequest(apiKey, apiVersion)) {
```

Comment:
While this means the benchmark is not exclusively tracking the costs of calling `onRequest` it does make the benchmark simulate the production load better.